tests/unit/adaptors/graph/test_manipulation.py

```diff
@@ -447,3 +447,23 @@ def test_raises_module_not_present_if_no_module(self):
 
         with pytest.raises(ModuleNotPresent):
             graph.squash_module("foo")
+
+    @pytest.mark.xfail(strict=True, reason="raises PanicException")
+    def test_correctly_handles_imports_within_descendants(self):
+        graph = ImportGraph()
+
+        graph.add_module("animals")
+        graph.add_module("food")
+        graph.add_import(importer="animals.dog", imported="food.chicken")
+        graph.add_import(importer="app.cli", imported="animals.dog")
+        # We want to check that this import within the descendants of `animals` does
+        # not cause problems. If this import is not properly removed then the imports map can
+        # become corrupted, since it will contain an import to modules that no longer exist.
+        # See https://github.com/seddonym/grimp/issues/195 for more details.
+        graph.add_import(importer="animals.dog", imported="animals.base")
```

Comment:
Should we add imports between deeper descendants too, such as grandchildren?

Reply:
We could. I'm not sure if it would add anything though, I think it shouldn't make any difference. So maybe it just confuses the test case to add more?